py4web/utils/form.py

```diff
@@ -535,9 +535,7 @@ def _sign_form(self):
         payload = {"ts": str(time.time())}
         if self.lifespan is not None:
             payload["exp"] = time.time() + self.lifespan
-        self.formkey = jwt.encode(payload, self._get_key(), algorithm="HS256").decode(
-            "utf-8"
-        )
+        self.formkey = jwt.encode(payload, self._get_key(), algorithm="HS256")
```

Comment:
Unfortunately this works in the test below but results in the wrong string being sent in the actual forms. For example, if you try login and explore the multipart form sent from browser to server you will get:

```
-----------------------------178483193414293891411069400419
Content-Disposition: form-data; name="_formkey"

b'eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ0cyI6IjE2MDg3Nzc1OTkuOTc4MjAyNiJ9.PhS9K39Rk-cBEqaAmMDhUztTePd0gpGK4-RdKAfYaPk'
-----------------------------178483193414293891411069400419--
```

I recommend leaving this line as is.

Reply:
Oops, I’ll check this and change the test. And I will investigate when the decode error occurs. Not always apparently. Which version of jwt are you using?